File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Workflow/WorkflowReferenceModel.cs
using System;
using System.Collections.Generic;
using VecompSoftware.DocSuiteWeb.Model.Entities.Commons;

namespace VecompSoftware.DocSuiteWeb.Model.Workflow
{
    public class WorkflowReferenceModel
    {
        #region [ Constructor ]
        public WorkflowReferenceModel()
        {
            Documents = new List<WorkflowReferenceBiblosModel>();
            DocumentUnits = new List<WorkflowReferenceDocumentUnitModel>();
        }
        #endregion

        #region [ Properties ]

        public Guid ReferenceId { get; set; }

        public DSWEnvironmentType ReferenceType { get; set; }

        public string ReferenceModel { get; set; }

        public string Title { get; set; }

        public ICollection<WorkflowReferenceBiblosModel> Documents { get; set; }

        public ICollection<WorkflowReferenceDocumentUnitModel> DocumentUnits { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/Workflows/WorkflowReferenceModel.ts
import DSWEnvironment = require('App/Models/Environment');
import WorkflowReferenceType = require('App/Models/Workflows/WorkflowReferenceType');
import WorkflowReferenceBiblosModel = require('./WorkflowReferenceBiblosModel');
import WorkflowReferenceDocumentUnitModel = require('App/Models/Workflows/WorkflowReferenceDocumentUnitModel');

interface WorkflowReferenceModel {
    ReferenceId: string;
    ReferenceType: DSWEnvironment;
    Title: string;
    ReferenceModel: string;
    WorkflowReferenceType: WorkflowReferenceType;
    Documents: WorkflowReferenceBiblosModel[];
    DocumentUnits: WorkflowReferenceDocumentUnitModel[];
}
export = WorkflowReferenceModel;